lib/galaxy/datatypes/tabular.py

```diff
@@ -20,17 +21,209 @@
+            for name, spec in dataset.metadata.spec.items():
+                if isinstance( spec.param, metadata.ColumnParameter ):
+                    try:
+                        i = int( getattr( dataset.metadata, name ) ) - 1
+                    except:
+                        i = -1
+                    if 0 <= i < columns and column_headers[i] is None:
+                        column_headers[i] = column_parameter_alias.get(name, name)
+
+            out.append( '<tr>' )
+            for i, header in enumerate( column_headers ):
+                out.append( '<th>' )
+                if header is None:
+                    out.append( column_number_format % str( i + 1 ) )
+                else:
+                    out.append( '%s.%s' % ( str( i + 1 ), escape( header ) ) )
+                out.append( '</th>' )
+            out.append( '</tr>' )
+        except Exception, exc:
+            log.exception( 'make_html_peek_header failed on HDA %s' % dataset.id )
+            raise Exception, "Can't create peek header %s" % str( exc )
+        return "".join( out )
+
+    def make_html_peek_rows( self, dataset, skipchars=None, **kwargs ):
+        if skipchars is None:
+            skipchars = []
+        out = []
+        try:
+            if not dataset.peek:
+                dataset.set_peek()
+            columns = dataset.metadata.columns
+            if columns is None:
+                columns = dataset.metadata.spec.columns.no_value
+            for line in dataset.peek.splitlines():
+                if line.startswith( tuple( skipchars ) ):
+                    out.append( '<tr><td colspan="100%%">%s</td></tr>' % escape( line ) )
+                elif line:
+                    elems = line.split( dataset.metadata.delimiter )
+                    # we may have an invalid comment line or invalid data
+                    if len( elems ) != columns:
+                        out.append( '<tr><td colspan="100%%">%s</td></tr>' % escape( line ) )
+                    else:
+                        out.append( '<tr>' )
+                        for elem in elems:
+                            out.append( '<td>%s</td>' % escape( elem ) )
+                        out.append( '</tr>' )
+        except Exception, exc:
+            log.exception( 'make_html_peek_rows failed on HDA %s' % dataset.id )
+            raise Exception, "Can't create peek rows %s" % str( exc )
+        return "".join( out )
+
+    def display_peek( self, dataset ):
+        """Returns formatted html of peek"""
+        return self.make_html_table( dataset )
+
+    # ------------- Dataproviders
+    @dataproviders.decorators.dataprovider_factory( 'column', dataproviders.column.ColumnarDataProvider.settings )
+    def column_dataprovider( self, dataset, **settings ):
+        """Uses column settings that are passed in"""
+        dataset_source = dataproviders.dataset.DatasetDataProvider( dataset )
+        delimiter = dataset.metadata.delimiter
```

Comment:
Chatted with @blankenberg about this - it seems that this is what ``no_value=`` does in the MetadataElement definition about. So I guess the ``no_value`` for ``delimiter`` should be ``\t``.